src/IO/WriteBufferFromHTTP.h

```diff
@@ -38,4 +37,43 @@ class WriteBufferFromHTTP : public WriteBufferFromOStream
     Poco::Net::HTTPResponse response;
 };
 
+class BuilderWriteBufferFromHTTP
+{
+    Poco::URI uri;
+    HTTPConnectionGroupType connection_group;
+    std::string method = Poco::Net::HTTPRequest::HTTP_POST; // POST or PUT only
+    bool bypass_proxy;
+    std::string content_type;
+    std::string content_encoding;
+    HTTPHeaderEntries additional_headers = {};
+    ConnectionTimeouts timeouts = {};
+    size_t buffer_size_ = DBMS_DEFAULT_BUFFER_SIZE;
+
+public:
+    explicit BuilderWriteBufferFromHTTP(const Poco::URI & uri_) : uri(uri_)
+    {}
+
+/// NOLINTBEGIN(bugprone-macro-parentheses)
+#define setterMember(name, member) \
+    BuilderWriteBufferFromHTTP & name(decltype(BuilderWriteBufferFromHTTP::member) arg_##member) \
+    { \
+        member = std::move(arg_##member); \
+        return *this; \
+    }
+
+    setterMember(withConnectionGroup, connection_group)
+    setterMember(withMethod, method)
+    setterMember(withBypassProxy, bypass_proxy)
+    setterMember(withContentType, content_type)
+    setterMember(withContentEncoding, content_encoding)
+    setterMember(bypassProxy, bypass_proxy)
+    setterMember(withAdditionalHeaders, additional_headers)
+    setterMember(withTimeouts, timeouts)
+    setterMember(withBufferSize, buffer_size_)
+#undef setterMember
+/// NOLINTEND(bugprone-macro-parentheses)
+
+    std::unique_ptr<WriteBufferFromHTTP> build();
```

Comment:
let's keep it consistent
```suggestion
    std::unique_ptr<WriteBufferFromHTTP> create();
```